### ISSUE TITLE
Question: Curious why only max output token counts are given rather than the precise count

### ISSUE THREAD

**Issue by titonova** [enricoros/big-AGI#591]
![image](https://github.com/enricoros/big-AGI/assets/41286822/5fc1cfc2-f6c2-4989-b49f-f8eb6cd26230)

I am just curious why the exact token count for the output messages aren't calculated as well, considering they are for the input tokens

**Comment by enricoros** [comment]
It's because that's the prediction before sending a message.

Before sending you don't know if the response will be 4 tokens or 400 or 4000. Hence we know the minimum that's going to cost (the input basically), and the maximum (input + full output) but we don't know how long the answer is going to be.

**Comment by titonova** [comment]
@enricoros I see, thanks. But why can't it be calculated *after* the message has been returned by the bot?

**Comment by enricoros** [comment]
@titonova yes for sure, but the prediction on the bottom is for before, while still typing it. In V 2.0.0 I'll display statistics for past messages.

**Comment by enricoros** [comment]
Closing as shipped.

Done as promised for v2. Open Message Info on any response and you get the exact input, output and reasoning token counts, cache reads and writes, the cost, and tok/s, all from the provider's usage report. The number in the composer stays an estimate, because before sending there is no answer to count.

@titonova this close comes from an automated backlog pass - if you disagree, feel free to reopen.

**Comment by titonova** [comment]
amazing work @enricoros
